tt_eager/tt_dnn/op_library/bmm/multi_core_reuse_optimized/bmm_op_multi_core_reuse_optimized.cpp

```diff
@@ -123,24 +123,15 @@ operation::ProgramWithCallbacks create_program(
     uint32_t start_core_y = 0;
     uint32_t num_cores_c = core_range.x;
     uint32_t num_cores_r = core_range.y;
+    uint32_t left_half_grid_x = num_cores_c > 1 ? num_cores_c / 2 : 1;
+    bool use_right_half_grid = num_cores_c > 1;
 
     bool split_half = num_cores_c > 2 && !in0_is_sharded;
     uint32_t half_core = split_half ? (num_cores_c) / 2 : num_cores_c - 1;
```

Comment:
Remove the split_half and half_core lines since you've replaced/renamed them.

Switch the order of your statements to:
```
bool use_right_half_grid = num_cores_c > 1;
uint32_t left_half_grid_x = use_right_half_grid ? num_cores_c / 2 : num_cores_c;
```

Reply:
i will put 1 instead of `num_cores_c` to be explicit in the `num_cores_c <= 1` case.

Reply:
I suggested to use num_cores_c in case the threshold changes for some reason